oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/prometheus/rule/Rules.java

```diff
@@ -47,29 +53,79 @@ public static List<Rule> loadRules(final String path) throws ModuleStartExceptio
     public static List<Rule> loadRules(final String path, List<String> enabledRules) throws ModuleStartException {
         File[] rules;
         try {
-            rules = ResourceUtils.getPathFiles(path);
+            rules = ResourceUtils.list(path);
         } catch (FileNotFoundException e) {
             throw new ModuleStartException("Load fetcher rules failed", e);
         }
-        return Arrays.stream(rules)
-            .filter(File::isFile)
-            .map(f -> {
-                try (Reader r = new FileReader(f)) {
-                    String fileName = f.getName();
-                    int dotIndex = fileName.lastIndexOf('.');
-                    fileName = (dotIndex == -1) ? fileName : fileName.substring(0, dotIndex);
-                    if (!enabledRules.contains(fileName)) {
-                        return null;
+        Map<String, Boolean> formedEnabledRules = enabledRules
+                .stream()
+                .map(rule -> {
+                    rule = rule.trim();
+                    if (rule.startsWith("/")) {
+                        rule = rule.substring(1);
+                    }
+                    if (rule.endsWith(".yaml")) {
+                        return rule;
+                    } else if (rule.endsWith(".yml")) {
+                        return rule.substring(0, rule.length() - 2) + "aml";
+                    } else {
+                        return rule + ".yaml";
                     }
-                    Rule rule = new Yaml().loadAs(r, Rule.class);
-                    rule.setName(fileName);
-                    return rule;
-                } catch (IOException e) {
-                    LOG.debug("Reading file {} failed", f, e);
+                })
+                .collect(Collectors.toMap(rule -> rule, $ -> false));
+
+        List<Rule> result = Arrays.stream(rules)
+                .flatMap(f -> {
+                    if (f.isDirectory()) {
+                        return Arrays.stream(Objects.requireNonNull(f.listFiles()))
+                                .filter(File::isFile)
+                                .map(file -> getRulesFromFile(formedEnabledRules, f, file));
+                    } else {
+                        return Stream.of(getRulesFromFile(formedEnabledRules, null, f));
+                    }
+                })
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+        if (formedEnabledRules.containsValue(false)) {
+            throw new UnexpectedException("Some configuration files of enabled rules are not found, enabled rules: " + formedEnabledRules.keySet());
+        }
+        return result;
+    }
+
+    private static Rule getRulesFromFile(Map<String, Boolean> formedEnabledRules, File directory, File file) {
+        try (Reader r = new FileReader(file)) {
+            String fileName = file.getName();
+            int dotIndex = fileName.lastIndexOf('.');
```

Comment:
You are using `lastIndexOf`, so `.asf.yaml` would only get the index of the dot in `.yaml`, isn't it?

Reply:
My fault. I forget it.